runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,311 @@
+
+|vim.keymap.set()| is a convenience function of API functions
+vim.api.|nvim_set_keymap()| and vim.api.|nvim_buf_set_keymap()|
+
+The first argument is a string or a table of strings containing the mode
+prefix for which the mapping will take effect: >lua
+
+	-- Normal mode
+	vim.keymap.set('n', '<Leader>ex1', '<Cmd>lua vim.notify("Example 1")<CR>')
+	-- Normal and Command-line mode
+	vim.keymap.set({'n', 'c'}, '<Leader>ex2', '<Cmd>lua vim.notify("Example 2")<CR>')
+
+The prefixes are the ones listed in |:map-arguments|, as well as "!" and and
+empty string "":
+
+| String value           | Help page     | Affected modes                           | Vim script equivalent |
+| ---------------------- | ------------- | ---------------------------------------- | --------------------- |
+| "" (empty string)      | |mapmode-nvo|   | Normal, Visual, Select, Operator-pending | |:map|                  |
+| `"n"`                    | |mapmode-n|     | Normal                                   | |:nmap|                 |
+| `"v"`                    | |mapmode-v|     | Visual and Select                        | |:vmap|                 |
+| `"s"`                    | |mapmode-s|     | Select                                   | |:smap|                 |
+| `"x"`                    | |mapmode-x|     | Visual                                   | |:xmap|                 |
+| `"o"`                    | |mapmode-o|     | Operator-pending                         | |:omap|                 |
+| `"!"`                    | |mapmode-ic|    | Insert and Command-line                  | |:map!|                 |
+| `"i"`                    | |mapmode-i|     | Insert                                   | |:imap|                 |
+| `"l"`                    | |mapmode-l|     | Insert, Command-line, Lang-Arg           | |:lmap|                 |
+| `"c"`                    | |mapmode-c|     | Command-line                             | |:cmap|                 |
+| `"t"`                    | |mapmode-t|     | Terminal                                 | |:tmap|                 |
+
+The second argument is the left-hand side of the mapping. An empty string is
+equivalent to `<Nop>`, which disables a key.
+
+The third argument is the right-hand side of the mapping, which can either be
+a string or a Lua function: >lua
+
+	vim.keymap.set('n', '<Leader>ex1', '<Cmd>echomsg "Example 1"<CR>')
+	vim.keymap.set('n', '<Leader>ex2', function() print("Example 2") end)
+	vim.keymap.set('n', '<Leader>pl1', require('plugin').plugin_action)
+	-- To avoid the startup cost of requiring the module, you can wrap it in a function to require it lazily when invoking the mapping:
+	vim.keymap.set('n', '<Leader>pl2', function() require('plugin').plugin_action() end)
+
+The fourth (optional) argument is a table containing boolean options for the
+mapping as defined in |:map-arguments|, and also `replace_keycodes`,
+`noremap`, `remap` and `desc`. >lua
```

Comment:
```suggestion
`remap` and `desc`. >lua
```

Reply:
(`vim.keymap.set` only uses `remap` (which is false by default, hence behaves as `Xnoremap`; to get `Xmap`, you need to pass `{remap = true}`)